tests/unit/PaymentErrorDialog.spec.ts

```diff
@@ -106,8 +106,39 @@ describe('Payment Error Dialog', () => {
       'We were unable to process your payment due to the following errors:'
     )
     expect(wrapper.findAll('p').at(2).text()).toContain('If this error persists')
-    expect(wrapper.findAll('li').length).toBe(1)
-    expect(wrapper.findAll('li').at(0).text()).toContain(padError[0].message)
+    expect(wrapper.findAll('li').length).toBe(0)
+    expect(wrapper.findAll('span').length).toBe(2)
+    expect(wrapper.findAll('span').at(1).text()).toContain('Your account is in the 3 day PAD confirmation period.')
+
+    expect(wrapper.find(RegistriesContactInfo).exists()).toBe(true)
+    expect(wrapper.find('#dialog-exit-button').exists()).toBe(true)
+
+    wrapper.destroy()
+  })
+
+  it('renders PAD warning messages correctly when they are present', () => {
+    store.state.stateModel.tombstone.authRoles = ['edit', 'view']
+    const wrapper = shallowMount(PaymentErrorDialog,
+      {
+        vuetify,
+        store,
+        propsData: { dialog: true, warnings: [
+          { warning: 'Test Warning 1' },
+          { warning: 'Test Warning 2' }
+        ]}
+      })
+
+    expect(wrapper.attributes('contentclass')).toBe('payment-error-dialog')
+    expect(wrapper.isVisible()).toBe(true)
+    expect(wrapper.find('#dialog-title').text()).toBe('Unable to Process Payment')
+    expect(wrapper.findAll('p').length).toBe(3)
+    expect(wrapper.findAll('p').at(0).text()).toContain('We are unable to process your payment')
+    expect(wrapper.findAll('p').at(1).text()).toContain('Please note the following warnings')
+    expect(wrapper.findAll('li').length).toBe(2)
+    expect(wrapper.findAll('li').at(0).text()).toBe('Test Warning 1')
+    expect(wrapper.findAll('li').at(1).text()).toBe('Test Warning 2')
+    expect(wrapper.find('#dialog-exit-button').exists()).toBe(true)
+    expect(wrapper.find('#dialog-okay-button').exists()).toBe(false)
```

Comment:
Added unit test for warning message. However, I am not sure it is warning.warning or warning.message. In create-ui, SaveErrorDialog.vue uses `warning.warning`, but PaymentErrorDialog.vue uses `warning.message`. They both use `saveWarnings` from the parent component App.vue. In edit-ui, both PaymentErrorDialog.vue and StaffPaymentErrorDialog.vue use`warning.message`, but SaveErrorDialog.vue uses `warning.warning`. They both use `saveWarnings` from the parent component. In all cases, there is no unit test for the warning message in the test file `PaymentErrorDialog.spec.ts`. Hence, I am not sure whether I should use `warning.message` or `warning.warning` for PaymentErrorDialog.

Reply:
@argush3 Can you help me take a look in BE whether the PaymentErrorDialog returns the warning message with the key `warning` or `message`? It is `warning.warning` while others `warning.message`.

Reply:
If you find that some UIs are incorrect, please create a new bug ticket to fix them.

In the meantime, if you don't get an answer right away for this UI, please revert the name change and update the unit test. (Ie, if it's broken now then it stays broken, but it's not blocking you.)

Reply:
I unresolved this conversation so we don't forget to answer the questions above and create bug tickets if needed.

Another way to handle this is to go ahead a create a new ticket now to ask the questions and to look at the affected UIs.

Reply:
Argus is checking the BE now and determine what's the correct response data.

Reply:
From looking at the code where the legal api calls the payment api to create the invoice when the filing is saved, I can see two main error paths.  I don't see anything with `warnings` in the legal api code from a quick search.

1. invoice is unable to be created due to timeout or connection issue when hitting pay api invoice endpoint.
legal api returns following response payload in this case
```
{
 ...
  "errors": [
    { 
      "message": "unable to create invoice for payment."
    }
  ]
}
```
2. pay api returns bad request response when pay api invoice endpoint invoked.
legal api returns following response payload in this case
```
{
  "errors": [
    {
      "payment_error_type": "error type…",
      "message": "error details…"
    }
  ]
}
```

Reply:
Thanks @argush3. We will use "message" as the key for the errors.

Reply:
@severinbeauvais Can you give feedback on the backend warnings issue?

Reply:
This dialog may be using generic error handling -- not only for payment errors.

Doesn't the BE sometimes return warnings when a filing is filed?

Reply:
> Thanks @argush3. We will use `message` as the key for the errors. Wait a second. If there is no `warnings` from BE response, then we can use either `warning` or `message` in the warnings array. FE now uses both `warning` and `message` in create-ui, edit-ui and filing-ui for warning messages. It will probably create inconsistent bugs in the future if BE includes warnings with `warning` as the key in the response.

I appreciate this discussion as we are trying to sort out whether there is a bug here or not, but this discussion has now completely de-railed this PR and ticket.

Let's take this discussion elsewhere -> create a ticket to investigate this and resolve it if needed. In THIS PR, only fix what was requested. In other words, leave the code as `warning.message`.

Reply:
Thanks Severin and Argus. We will leave the code as `warning.message`.

Reply:
What are we doing to keep track of the possible warnings bug? Is there a ticket number?

Reply:
I'm not sure. I don't know how to create a ticket. Can you give me a walkthrough for creating a ticket this issue?

Reply:
Please add new ticket number to 9014. No need to respond to this conversation. I'm closing it now.